mesa/experimental/devs/eventlist.py

```diff
@@ -0,0 +1,191 @@
+import itertools
+from enum import IntEnum
+from heapq import heapify, heappop, heappush
+from types import MethodType
+from typing import Any, Callable
+from weakref import WeakMethod, ref
+
+
+class InstanceCounterMeta(type):
+    """Metaclass to make instance counter not share count with descendants
```

Comment:
Why is it desired that the counter is not shared with descendants? This is used for the unique_id, which, as the name suggests, should be unique. Not sharing the counter might break this